cli/tools/lint/linter.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::path::Path;
+
+use deno_ast::MediaType;
+use deno_ast::ModuleSpecifier;
+use deno_ast::ParsedSource;
+use deno_ast::SourceTextInfo;
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use deno_graph::ModuleGraph;
+use deno_lint::diagnostic::LintDiagnostic;
+use deno_lint::linter::LintConfig as DenoLintConfig;
+use deno_lint::linter::LintFileOptions;
+use deno_lint::linter::Linter as DenoLintLinter;
+use deno_lint::linter::LinterOptions;
+
+use crate::util::fs::atomic_write_file_with_retries;
+use crate::util::fs::specifier_from_file_path;
+
+use super::rules::FileOrPackageLintRule;
+use super::rules::PackageLintRule;
+use super::ConfiguredRules;
+
+pub struct CliLinterOptions {
+  pub configured_rules: ConfiguredRules,
+  pub fix: bool,
+  pub deno_lint_config: DenoLintConfig,
+}
+
+#[derive(Debug)]
+pub struct CliLinter {
+  fix: bool,
+  package_rules: Vec<Box<dyn PackageLintRule>>,
+  linter: DenoLintLinter,
+  deno_lint_config: DenoLintConfig,
+}
+
+impl CliLinter {
+  pub fn new(options: CliLinterOptions) -> Self {
+    let rules = options.configured_rules.rules;
+    let mut deno_lint_rules = Vec::with_capacity(rules.len());
+    let mut package_rules = Vec::with_capacity(rules.len());
+    for rule in rules {
+      match rule.into_file_or_pkg_rule() {
+        FileOrPackageLintRule::File(rule) => {
+          deno_lint_rules.push(rule);
+        }
+        FileOrPackageLintRule::Package(rule) => {
+          package_rules.push(rule);
+        }
+      }
+    }
+    Self {
+      fix: options.fix,
+      package_rules,
+      linter: DenoLintLinter::new(LinterOptions {
+        rules: deno_lint_rules,
+        all_rule_codes: options.configured_rules.all_rule_codes,
+        custom_ignore_file_directive: None,
+        custom_ignore_diagnostic_directive: None,
+      }),
+      deno_lint_config: options.deno_lint_config,
+    }
+  }
+
+  pub fn has_package_rules(&self) -> bool {
+    !self.package_rules.is_empty()
+  }
+
+  pub fn lint_package(
```

Comment:
This is also incrementally working towards having these lint rules show up in the lsp.